src/kbmod/work_unit.py

```diff
@@ -1116,9 +1130,12 @@ def read_image_data_from_hdul(hdul, idx):
     # Get the variance layer.
     var = hdul[f"VAR_{idx}"].data.astype(np.single)
 
-    # Allow the mask to be optional. Use an empty mask if none is present.
+    # Allow the mask to be optional. Apply the mask if it is present
+    # and use an empty mask if there is no mask layer.
     if f"MSK_{idx}" in hdul:
         mask = hdul[f"MSK_{idx}"].data.astype(np.single)
+        sci[mask > 0] = np.nan
+        var[mask > 0] = np.nan
```

Comment:
is it ok that all of the masked pixels in science and variance are converted to `np.nan`? if there are pixels masked out for reasons other than their being NaNs we'd lose that data. doesn't matter for search but could cause problems for debugging/provenance

Reply:
I think it is okay, but would like you to check my logic:
- We apply the masking flags in the standardizer, so we only propagate those bits we want to capture.
- We are already binarizing the mask when saving it (the write function saves "mask > 0" instead of the raw integers.
- We always apply OxFFFFFFF for the mask in the code.